contracts/contracts/strategies/NativeStaking/NativeStakingSSVStrategy.sol

```diff
@@ -0,0 +1,322 @@
+        address _wethAddress,
+        address _ssvToken,
+        address _ssvNetwork,
+        uint256 _maxValidators,
+        address _feeAccumulator,
+        address _beaconChainDepositContract
+    )
+        InitializableAbstractStrategy(_baseConfig)
+        ValidatorAccountant(
+            _wethAddress,
+            _baseConfig.vaultAddress,
+            _beaconChainDepositContract,
+            _ssvNetwork,
+            _maxValidators
+        )
+    {
+        SSV_TOKEN_ADDRESS = _ssvToken;
+        FEE_ACCUMULATOR_ADDRESS = payable(_feeAccumulator);
+    }
+
+    /// @notice initialize function, to set up initial internal state
+    /// @param _rewardTokenAddresses Address of reward token for platform
+    /// @param _assets Addresses of initial supported assets
+    /// @param _pTokens Platform Token corresponding addresses
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /// @dev Convert accumulated ETH to WETH and send to the Harvester.
+    /// Will revert if the strategy is paused for accounting.
+    function _collectRewardTokens() internal override whenNotPaused {
+        // collect ETH from execution rewards from the fee accumulator
+        uint256 executionRewards = FeeAccumulator(FEE_ACCUMULATOR_ADDRESS)
+            .collect();
+
+        // total ETH rewards to be harvested = execution rewards + consensus rewards
+        uint256 ethRewards = executionRewards + consensusRewards;
+
+        require(
+            address(this).balance >= ethRewards,
+            "insufficient eth balance"
+        );
+
+        if (ethRewards > 0) {
+            // reset the counter keeping track of beacon chain consensus rewards
+            consensusRewards = 0;
+
+            // Convert ETH rewards to WETH
+            IWETH9(WETH_TOKEN_ADDRESS).deposit{ value: ethRewards }();
+
+            emit RewardTokenCollected(
+                harvesterAddress,
+                WETH_TOKEN_ADDRESS,
+                ethRewards
+            );
+            IERC20(WETH_TOKEN_ADDRESS).safeTransfer(
+                harvesterAddress,
+                ethRewards
+            );
+        }
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just checks the asset is WETH and emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _asset Address of asset to deposit. Has to be WETH.
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
```

Comment:
Since the vault no longer calls `deposit`, could we make this function revert, and then we can remove the code around `depositedWethAccountedFor` here and in `depositAll`?

Reply:
There's a design decision on whether the Native Staking Strategy should be the default strategy
https://www.notion.so/originprotocol/Should-the-Native-Staking-strategy-be-the-default-for-WETH-718c1b9a43154b13a05caee97e8a75d6